app/services/foreman_rh_cloud/ui_request_forwarder.rb

```diff
@@ -0,0 +1,106 @@
+require 'rest-client'
+
+module ForemanRhCloud
+  class UIRequestForwarder
+    include ForemanRhCloud::GatewayRequest
+
+    def forward_request(original_request, path, controller_name, user, organization, location)
+      TagsAuth.new(user, logger).update_tag if scope_request?(original_request)
+
+      forward_params = prepare_forward_params(original_request, user: user, organization: organization, location: location).to_a
+      logger.debug("Request parameters for UI request: #{forward_params}")
+
+      forward_payload = prepare_forward_payload(original_request, controller_name)
+
+      logger.debug("User agent for UI is: #{http_user_agent(original_request)}")
+
+      request_opts = prepare_request_opts(original_request, path, forward_payload, forward_params)
+
+      logger.debug("Sending request to: #{request_opts[:url]}")
+
+      execute_cloud_request(request_opts)
```

Comment:
I'd highly recommend to have an explicit list of APIs that are allowed to be forwarder. The list can contain the config on `tag` v. `tags`.

Reply:
For now I didn't find any instances of `tag`, only `tags`. If we get this kind of inconsistency, we will deal with it